tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -263,6 +263,49 @@ function testRuntimeTypeNameOfIntersectionType() {
 
 type Error error<record { string message; }>;
 
+type ReadonlyTypeDef3 readonly;
+
+type ReadonlyObjectType1 object {
+    int atb1;
+    function call() returns int;
+};
+
+type IntersectionType1 ReadonlyTypeDef3 & record {|
+    int a;
+|};
+
+type FooType record {|
+    int a;
+|};
+
+type IntersectionType2 ReadonlyTypeDef3 & FooType;
+type IntersectionType3 FooType & ReadonlyTypeDef3;
+type IntersectionType4 ReadonlyTypeDef3 & ReadonlyObjectType1;
+type IntersectionType5 ReadonlyObjectType1 & ReadonlyTypeDef3;
```

Comment:
Shall we add test with 
1. three intersection and two members are readonly types
2. Multiple intersection with error intersection(with details) since we allow intersection between errors.

Reply:
Fixed in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/4746e1e627f5b2dd805496dc33e6aeb2fb81f730).